lib/common/memory.c

```diff
@@ -152,11 +154,9 @@ void *h2o_mem_alloc_pool(h2o_mem_pool_t *pool, size_t sz)
     }
 
     /* return a valid pointer even for 0 sized allocs */
-    if (sz == 0)
-        sz = 1;
+    if (H2O_UNLIKELY(sz == 0))
+        sz = alignment;
```

Comment:
I think that `h2o_mem_alloc_pool(int, 0)` should return a unique pointer that is aligned for `int`. Hence keep the code as-is.

Reply:
you mean  leave `sz = 1` as-is? 
seems we can do this.

even with `sz = alignment`,  `h2o_mem_alloc_pool(int, 0)` also return a unique pointer that is aligned for int.

Reply:
Thank you for the changes.

FTR the issue was that there was a case that left the `chunk_offset` unchanged leading to the uniqueness not being guaranteed; for example when `chunk_offset` is a multiple of 4 and `h2o_mem_alloc_pool(pool, uint32_t, 0)` is called.